plugins/inputs/prometheus/kubernetes.go

```diff
@@ -0,0 +1,201 @@
+package prometheus
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/url"
+	"os/user"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/ericchiang/k8s"
+	corev1 "github.com/ericchiang/k8s/apis/core/v1"
+	"gopkg.in/yaml.v2"
+)
+
+type payload struct {
+	eventype string
+	pod      *corev1.Pod
+}
+
+// loadClient parses a kubeconfig from a file and returns a Kubernetes
+// client. It does not support extensions or client auth providers.
+func loadClient(kubeconfigPath string) (*k8s.Client, error) {
+	data, err := ioutil.ReadFile(kubeconfigPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed reading '%s': %v", kubeconfigPath, err)
+	}
+
+	// Unmarshal YAML into a Kubernetes config object.
+	var config k8s.Config
+	if err := yaml.Unmarshal(data, &config); err != nil {
+		return nil, err
+	}
+	return k8s.NewClient(&config)
+}
+
+func (p *Prometheus) start(ctx context.Context) error {
+	client, err := k8s.NewInClusterClient()
+	if err != nil {
+		u, err := user.Current()
+		if err != nil {
+			return fmt.Errorf("Failed to get current user - %v", err)
+		}
+		configLocation := filepath.Join(u.HomeDir, ".kube/config")
+		if p.KubeConfig != "" {
+			configLocation = p.KubeConfig
+		}
+		client, err = loadClient(configLocation)
+		if err != nil {
+			return err
+		}
+	}
+
+	p.wg = sync.WaitGroup{}
+	in := make(chan payload)
+
+	p.wg.Add(1)
+	go func() {
+		defer p.wg.Done()
+		for {
+			select {
+			case <-ctx.Done():
+				break
```

Comment:
Need to use a return or we won't exit the for loop

Reply:
should we remove the err checking logic on the p.watch as well, since it doesn't effectively do anything for us?

Reply:
I would probably move the logging from line 84 up to this function, and just `return err` in the watch function.  I see that there is both an error and debug level message in watch, but since they both terminate the watch function I think they should both be error level.